src/org/loklak/http/ClientConnection.java

```diff
@@ -241,44 +244,127 @@ private void init() throws IOException {
             throw new IOException("client connection to " + this.request.getURI() + " fail: no connection");
         }
     }
-    
+
+    public static CloseableHttpClient getCustomClosableHttpClient(boolean useAuthentication) {
+        return HttpClients.custom()
+                .setConnectionManager(getConnctionManager(useAuthentication))
+                .setDefaultRequestConfig(defaultRequestConfig)
+                .build();
+    }
+
+    /**
+     * Get body of HttpEntity as String
+     * @param httpREntity
+     * @return response body
+     * @throws IOException when readLine fails
+     */
+    public static String getHTML(HttpEntity httpEntity) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(httpEntity.getContent()));
+        String content = null;
+        while ((content = reader.readLine()) != null)
+            sb.append(content);
+        return sb.toString();
+    }
+
+    /**
+     * Get redirect URL from Meta tag in response body
+     * @param httpEntity
+     * @return redirect URL if any else null
+     * @throws IOException when getHTML() fails
+     */
+    public static String getMetaRedirectURL(HttpEntity httpEntity) throws IOException {
+        String html = getHTML(httpEntity);
+        html = html.replace("\n", "");  // May have a line break between tags
+        if (html.length() == 0)
+            return null;
+        int indexHttpEquiv = html.indexOf("http-equiv=\"refresh\"");
+        if (indexHttpEquiv < 0)
+            return null;
+        html = html.substring(indexHttpEquiv);
+        int indexContent = html.indexOf("content=");
+        if (indexContent < 0)
+            return null;
+        html = html.substring(indexContent);
+        int indexURLStart = html.indexOf("http");
+        if (indexURLStart < 0)
+            return null;
+        html = html.substring(indexURLStart);
+        int indexURLEnd = html.indexOf("\"");
+        if (indexURLEnd < 0)
+            return null;
```

Comment:
It will give the correct results but will be slower. This method is not hard to follow so there will be no problem in maintaining it.